src/statement-list/statement_list_typecheck.cpp

```diff
@@ -1296,23 +1322,58 @@ void statement_list_typecheckt::typecheck_called_function_block(
 
 exprt statement_list_typecheckt::typecheck_function_call_arguments(
   const std::vector<equal_exprt> &assignments,
-  const code_typet::parametert &param)
+  const code_typet::parametert &param,
+  const symbolt &tia_element)
+{
+  for(const equal_exprt &assignment : assignments)
+  {
+    const symbol_exprt &lhs{to_symbol_expr(assignment.lhs())};
+    if(param.get_base_name() == lhs.get_identifier())
+    {
+      const symbol_exprt &rhs{to_symbol_expr(assignment.rhs())};
+      const exprt assigned_variable{
+        typecheck_identifier(tia_element, rhs.get_identifier())};
+      if(param.type() == assigned_variable.type())
+        return assigned_variable;
+      else
+      {
+        error() << "Types of parameter assignment do not match: "
+                << param.type().id() << " != " << assigned_variable.type().id()
+                << eom;
+        throw TYPECHECK_ERROR;
+      }
+    }
+  }
+  error() << "No assignment found for function parameter "
+          << param.get_identifier() << eom;
+  throw TYPECHECK_ERROR;
+}
+
+exprt statement_list_typecheckt::typecheck_return_value_assignment(
+  const std::vector<equal_exprt> &assignments,
+  const typet &return_type,
+  const symbolt &tia_element)
 {
   for(const equal_exprt &assignment : assignments)
   {
     const symbol_exprt &lhs{to_symbol_expr(assignment.lhs())};
-    if(param.get_identifier() == lhs.get_identifier())
+    if(RET_VAL_NAME == id2string(lhs.get_identifier()))
```

Comment:
Change `RET_VAL_NAME` to `ID_xxx` so we don't have to perform a string comparison here.